MediaBrowser.Providers/MediaInfo/AudioFileProber.cs

```diff
@@ -303,48 +271,104 @@ private async Task FetchDataFromTags(Audio audio, Model.MediaInfo.MediaInfo medi
             {
-                audio.SetProviderId(MetadataProvider.MusicBrainzAlbumArtist, tags.MusicBrainzReleaseArtistId);
+                track.AdditionalFields.TryGetValue("MUSICBRAINZ_ALBUMARTISTID", out var musicBrainzReleaseArtistIdTag);
+                if (musicBrainzReleaseArtistIdTag is null)
+                {
+                    track.AdditionalFields.TryGetValue("MusicBrainz Album Artist Id", out musicBrainzReleaseArtistIdTag);
+                }
+
+                if (musicBrainzReleaseArtistIdTag is not null)
+                {
+                    audio.SetProviderId(MetadataProvider.MusicBrainzAlbumArtist, musicBrainzReleaseArtistIdTag);
+                }
             }
 
             if (options.ReplaceAllMetadata || !audio.TryGetProviderId(MetadataProvider.MusicBrainzAlbum, out _))
             {
-                audio.SetProviderId(MetadataProvider.MusicBrainzAlbum, tags.MusicBrainzReleaseId);
+                track.AdditionalFields.TryGetValue("MUSICBRAINZ_ALBUMID", out var musicBrainzReleaseIdTag);
+                if (musicBrainzReleaseIdTag is null)
+                {
+                    track.AdditionalFields.TryGetValue("MusicBrainz Album Id", out musicBrainzReleaseIdTag);
+                }
+
+                if (musicBrainzReleaseIdTag is not null)
+                {
+                    audio.SetProviderId(MetadataProvider.MusicBrainzAlbum, musicBrainzReleaseIdTag);
+                }
             }
 
             if (options.ReplaceAllMetadata || !audio.TryGetProviderId(MetadataProvider.MusicBrainzReleaseGroup, out _))
             {
-                audio.SetProviderId(MetadataProvider.MusicBrainzReleaseGroup, tags.MusicBrainzReleaseGroupId);
+                track.AdditionalFields.TryGetValue("MUSICBRAINZ_RELEASEGROUPID", out var musicBrainzReleaseGroupIdTag);
+                if (musicBrainzReleaseGroupIdTag is null)
+                {
+                    track.AdditionalFields.TryGetValue("MusicBrainz Release Group Id", out musicBrainzReleaseGroupIdTag);
+                }
+
+                if (musicBrainzReleaseGroupIdTag is not null)
+                {
+                    audio.SetProviderId(MetadataProvider.MusicBrainzReleaseGroup, musicBrainzReleaseGroupIdTag);
+                }
             }
 
             if (options.ReplaceAllMetadata || !audio.TryGetProviderId(MetadataProvider.MusicBrainzTrack, out _))
             {
-                // Fallback to ffprobe as TagLib incorrectly provides recording MBID in `tags.MusicBrainzTrackId`.
-                // See https://github.com/mono/taglib-sharp/issues/304
-                var trackMbId = mediaInfo.GetProviderId(MetadataProvider.MusicBrainzTrack);
+                track.AdditionalFields.TryGetValue("MUSICBRAINZ_RELEASETRACKID", out var trackMbId);
+                if (trackMbId is null)
+                {
+                    track.AdditionalFields.TryGetValue("MusicBrainz Release Track Id", out trackMbId);
+                }
+
```

Comment:
```suggestion
                if ((track.AdditionalFields.TryGetValue("MUSICBRAINZ_RELEASETRACKID", out var trackMbId)
                     || track.AdditionalFields.TryGetValue("MusicBrainz Release Track Id", out trackMbId))
                    && !string.IsNullOrEmpty(trackMbId))
                {
                    audio.SetProviderId(MetadataProvider.MusicBrainzTrack, trackMbId);
                }
```

with the next few lines as well.

These changes are mostly cosmetic, but I think it shows clearer intent when quickly reading the code